libs/ledger-live-common/src/device/use-cases/appDataBackup/backupAppDataUseCase.ts

```diff
@@ -0,0 +1,63 @@
+import { Observable, switchMap } from "rxjs";
+import {
+  AppName,
+  AppStorageType,
+  BackupAppDataError,
+  BackupAppDataEvent,
+  BackupAppDataEventType,
+  StorageProvider,
+} from "./types";
+import { AppStorageInfo } from "@ledgerhq/device-core/index";
+import { DeviceModelId } from "@ledgerhq/devices";
+
+/**
+ * Backs up the application data for a specific app on a Ledger device. All interactions
+ * with the storage provider are handled by this function.
+ *
+ * @param storageProvider - The storage provider object used for storing the backup data.
+ * @param appName - The name of the application to backup.
+ * @param backupAppDataFn - The function that returns observable for the backup process.
+ * @returns An observable that emits BackupAppDataEvent during the backup process.
+ * @throws {BackupAppDataError}
+ */
+export function backupAppDataUseCase(
+  appName: AppName,
+  deviceModelId: DeviceModelId,
+  storageProvider: StorageProvider<AppStorageType>,
+  backupAppDataFn: () => Observable<BackupAppDataEvent>,
+): Observable<BackupAppDataEvent> {
+  let appDataInfo: AppStorageInfo;
+  const obs: Observable<BackupAppDataEvent> = backupAppDataFn().pipe(
+    switchMap(async (event: BackupAppDataEvent): Promise<BackupAppDataEvent> => {
```

Comment:
[ASK] Why is switchMap returning a `Promise<BackupDataEvent>` ? I believe you should return an `Observable<BackupDataEvent>` using `of` or something like that, and probably `throwError` as well instead of just throwing as you are in an observable

Reply:
It seems my previous comment was wrong, there is interoperability with promises / async function (although not very well documented). You can disregard the previous comment but i still believe that staying on one paradygm (either Observable  or Promises) is better.